File: scala/src/main/scala/day1/SolutionB.scala
package day1

import java.nio.file.{Files, Paths}

object SolutionB extends App {

  val input = Files.readString(Paths.get("src/main/resources/day1/input_a.txt"))
  val groups = input.split("\n\n").map(_.split("\n").map(_.toInt))
  val calories = groups.map(_.sum).sorted(Ordering[Int].reverse).take(3).sum

  println(calories) // 207456
}

File: python/aoc_2022/day1/solution_b.py
if __name__ == "__main__":
    
    with open("inputs/day1/input_a.txt", "r") as f:
        input = f.read()
        
    groups = map(lambda group: map(lambda line: int(line), group), map(lambda group: group.split("\n"), input.split("\n\n")))    
    group_sums = map(lambda group: sum(group), groups)
    result = sum(sorted(group_sums, reverse=True)[:3])
    print(result) # 207456
    